march_rqt_gait_generator/test/rostests/gait_generator_controller_test

```diff
@@ -0,0 +1,361 @@
+#!/usr/bin/env python
+import unittest
+import rospy
+import rostest
+
+from mock import Mock
+import rospkg
+from urdf_parser_py import urdf
+
+
+from march_rqt_gait_generator.gait_generator_controller import GaitGeneratorController
+from march_rqt_gait_generator.model.modifiable_subgait import ModifiableSubgait
+
+PKG = 'march_rqt_gait_generator'
+
+class GaitGeneratorControllerTest(unittest.TestCase):
+    # def __init__(self, *args):
+    #     super(GaitGeneratorControllerTest, self).__init__(*args)
+    #     rospy.init_node("test_gait_generator", anonymous=True)
+
+    def setUp(self):
+        rospy.init_node('gait_generator_controller_test', anonymous=True)
+        self.gait_generator_view = Mock()
+        self.gait_generator_view.topic_name_line_edit.text = Mock(return_value='/march/my_fancy_topic')
+        self.gait_generator_view.joint_widgets.__getitem__ = Mock()
+        self.gait_name = 'walk'
+        self.subgait_name = 'left_swing'
+        self.version = 'MV_walk_leftswing_v2'
+        self.resources_folder = rospkg.RosPack().get_path('march_rqt_gait_generator') + '/test/resources'
+        self.robot = urdf.Robot.from_xml_file(rospkg.RosPack().get_path('march_description') + '/urdf/march4.urdf')
+        self.subgait_path = '{rsc}/{gait}/{subgait}/{version}.subgait'.format(rsc=self.resources_folder,
+                                                                              gait=self.gait_name,
+                                                                              subgait=self.subgait_name,
+                                                                              version=self.version)
+        self.gait_generator_controller = GaitGeneratorController(self.gait_generator_view)
+        self.duration = self.gait_generator_controller.subgait.duration
+        self.num_joints = len(self.gait_generator_controller.subgait.joints)
```

Comment:
If you remove the `view` dependency from the `GaitGeneratorController` then you can remove the mocks and ros node for testing the controller, which makes this a lot simpler and easier to refactor